model/DataObject.php

```diff
@@ -398,10 +398,15 @@ public static function composite_fields($class = null, $aggregated = true) {
 	 * field values.  Normally this contructor is only used by the internal systems that get objects from the database.
 	 * @param boolean $isSingleton This this to true if this is a singleton() object, a stub for calling methods.
 	 *                             Singletons don't have their defaults set.
+	 * @param DataModel $model
+	 * @param array $queryParams List of DataQuery params necessary to lazy load, or load related objects.
 	 */
-	public function __construct($record = null, $isSingleton = false, $model = null) {
+	public function __construct($record = null, $isSingleton = false, $model = null, $queryParams = array()) {
```

Comment:
Yes, but it's necessary because otherwise lazy-loading blows up during dependency injection. VirtualPage cost me at least 2 hours debugging due to this issue; DI will request various properties from an object, and VirtualPage nearly always loads the virtualised page during these.

The only way to fix this is to assign query parameters at object construction, so they are always available.